new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -54,4 +111,83 @@ public static String createUniqueNamespace() {
         }
     }
 
+    public static List<String> listNamespaces() {
+        ArrayList<String> nameSpaces = new ArrayList<>();
+        try {
+            V1NamespaceList namespaceList = coreV1Api.listNamespace(pretty, allowWatchBookmarks, null, null, null, null, resourceVersion, timeoutSeconds, false);
+
+            for (V1Namespace namespace : namespaceList.getItems()) {
+                nameSpaces.add(namespace.getMetadata().getName());
+            }
+        } catch (ApiException e) {
+            e.printStackTrace();
+        }
+
+        return nameSpaces;
+    }
+
+    public static boolean deleteNamespace(String name) {
+        V1DeleteOptions deleteOptions = new V1DeleteOptions();
+        try {
+            V1Status status = coreV1Api.deleteNamespace(name, pretty, null, timeoutSeconds, false, "Foreground", deleteOptions);
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    // --------------------------- Custom Resource Domain -----------------------------------
+
+    public static boolean createDomain(String domainUID, String namespace, String domainYAML) {
+        final String localVarPath =
+                DOMAIN_PATH.replaceAll("\\{namespace\\}", apiClient.escapeString(namespace));
+
+        Object json = null;
+        try {
+            json = convertYamlToJson(domainYAML);
+            Object response = customObjectsApi.createClusterCustomObject(DOMAIN_GROUP, DOMAIN_VERSION, localVarPath, json, null);
+            return true;
+        } catch (ApiException | IOException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    private static Object convertYamlToJson(String yamlFile) throws IOException {
+        ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+        Object yamlObj = yamlReader.readValue(new File(yamlFile), Object.class);
+        System.out.println("Kubernetes.convertYamlToJson yaml: " + yamlObj);
+
+        ObjectMapper jsonWriter = new ObjectMapper();
+        String writeValueAsString = jsonWriter.writeValueAsString(yamlObj);
+        System.out.println("Kubernetes.convertYamlToJson writeValueAsString: " + writeValueAsString);
+        JsonNode root = new ObjectMapper().readTree(writeValueAsString);
+        return root;
+    }
+
+    public static List<String> listDomains(String namespace) {
+        ArrayList<String> domains = new ArrayList<>();
+        try {
+            Map response = (Map) customObjectsApi.listNamespacedCustomObject(DOMAIN_GROUP, DOMAIN_VERSION, namespace, DOMAIN_PLURAL, null,
```

Comment:
nit pick - put each param on a new line - maybe put a comment after them to say what they are, since the six nulls seem a bit magical :)

Reply:
Yeah, it's too bad they don't have a builder for requests (I'll validate that this is still correct).

Reply:
split params to individual lines and added comments to explain parameter meaning.